hathor/sysctl/protocol.py

```diff
@@ -0,0 +1,111 @@
+        self.root = root
+
+    def lineReceived(self, raw: bytes) -> None:
+        try:
+            line = raw.decode('utf-8').strip()
+        except UnicodeDecodeError:
+            self.sendError('command is not utf-8 valid')
+        if line == '!backup':
+            self.backup()
+            return
+        head, separator, tail = line.partition('=')
+        head = head.strip()
+        tail = tail.strip()
+        if separator == '=':
+            self.set(head, tail)
+        else:
+            self.get(head)
+
+    def sendError(self, msg: str) -> None:
+        """Send an error message to the client. Used when a command fails."""
+        self.sendLine(f'[error] {msg}'.encode('utf-8'))
+
+    def set(self, path: str, value_str: str) -> None:
+        """Run a `set` command in sysctl."""
+        try:
+            value = self._deserialize(value_str)
+        except json.JSONDecodeError:
+            self.sendError('value: not a valid json')
+            return
+        except ValueError:
+            self.sendError('value: wrong format')
+            return
+
+        try:
+            self.root.set(path, value)
+        except SysctlEntryNotFound:
+            self.sendError(f'{path} not found')
+        except SysctlReadOnlyEntry:
+            self.sendError(f'cannot write to {path}')
+        except SysctlException as e:
+            self.sendError(str(e))
+        except TypeError as e:
+            self.sendError(str(e))
+
+    def get(self, path: str) -> None:
+        """Run a `get` command in sysctl."""
+        try:
+            value = self.root.get(path)
+        except SysctlEntryNotFound:
+            self.sendError(f'{path} not found')
+        except SysctlWriteOnlyEntry:
+            self.sendError(f'cannot read from {path}')
+        else:
+            output = self._serialize(value)
+            self.sendLine(output.encode('utf-8'))
+
+    def backup(self) -> None:
+        """Run a `backup` command, sending all parameters to the client."""
+        for key, value in self.root.get_all():
+            output = f'{key}={self._serialize(value)}'
+            self.sendLine(output.encode('utf-8'))
+
+    def _serialize(self, value: Any) -> str:
+        """Serialize the return of a sysctl getter."""
+        output: str
+        if isinstance(value, tuple):
+            parts = (json.dumps(x) for x in value)
+            output = f'({", ".join(parts)})'
+        else:
+            output = json.dumps(value)
+        return output
+
+    def _deserialize(self, value_str: str) -> Any:
+        """Deserialize a value sent by the client."""
+        if len(value_str) == 0:
+            return ()
+        if value_str[0] == '(':
+            if value_str[-1] != ')':
+                raise ValueError('not closing (')
+            parts = [x.strip() for x in value_str[1:-1].split(',')]
+            return tuple(json.loads(x) for x in parts)
+        return json.loads(value_str)
```

Comment:
What if an user sends `my.values=1, 2, 3)`? Maybe you could just remove support for parenthesis, it would be simpler to implement and similar to use:

```
p2p.rate_limit.global.send_tips=20, 4
p2p.rate_limit.global.send_tips
20, 4
```

Reply:
I agree. Done here: https://github.com/HathorNetwork/hathor-core/pull/587/commits/a7d22235a62e2de889a3b1765e011f2bc0341453